### DISCUSSION TITLE
Connecting to Spanet

### DISCUSSION THREAD

**Discussion by vdbpeter** [wayne-love/sn_esp32#15] [Q&A]
I have a Spanet SV 3 in my newly installed spa. Am very interested in what you're doing, along with scuznuts' work.

Am just in the phase of building and testing an ESP32 with various images.

From the discussions, I assume we connect to the RJ45 style connectors  but no one has actually specified which port... I don't like making assumptions with these sorts of things!

Wayne's diagram is helpful but can you add more detail to it please? (To remove all ambiguity)

Thanks

**Comment by vdbpeter** [comment]
![20221224_185912](https://user-images.githubusercontent.com/32290447/212024844-af997e8c-dc67-414c-9809-ab8ffd8b9d0a.jpg)

**Comment by wayne-love** [comment]
The module that I designed replaces the Spanet wifi module so plugs into the same port.  Note, there is a chance that the firmware on the controller has been updated to work with the Ver3 module and that the serial port settings or the serial protocol that is used.  I would be surprised if they have changed the physical characteristics of the port so you should be safe (this is only my guess so as always proceed with caution :-) )

BTW any chance of cracking open the V3 wifi module and taking a photo of the insides?

**Comment by wayne-love** [comment]
Sorry just reread your post, confused the controller version with the wifi module version.  Unplug the wifi module and esp32 plugs into the same port.

**Reply by vdbpeter** [reply]
I don't have a wifi unit, and want to minimise the guess work.

Judging by my photo, EXP1 looks like a candidate for the ESP?

I agree they're unlikely to radically change the protocol or command set.

So, to be clear, the Controller rj45 physical ports host tx, rx, +v and ground? The tx and rx voltage levels- are they TTL? 0v to what voltage? What are the other pins used for?

And the serial port rate is 115200?

Cheers

**Comment by wayne-love** [comment]
Baud rate is 115200.Pins 1/2 or 7/8 are 12v power, the other is 12v signaling (drives the colour of the status led on the side of the spa). Check my circuit diagram and it will give you a clue as to which are which.  Pins 3/4 are unused.  5/6 are serial data, TTL 3.3v.EXP1 is the port to be used, the cable in it needs to be unplugged which will disable the health led on the side of the spa.  At the moment my circuit does not have a pass through like the genuine wifi unit does so this is on the todo list.On 12 Jan 2023, at 9:07 pm, vdbpeter ***@***.***> wrote:﻿
I don't have a wifi unit, and want to minimise the guess work.
I agree they're unlikely to radically change the protocol or command set.
So, to be clear, the Controller rj45 physical ports host tx, rx, +v and ground? The tx and rx voltage levels- are they TTL? 0v to what voltage? What are the other pins used for?
And the serial port rate is 115200?
Cheers

**Reply by vdbpeter** [reply]
Thanks, but please see my other reply. 
Mine doesnt have the Status LED plugged in, but checking out connector "HB S7", that has +14v on Pin1 and what seems to be a FET pulling Pin2 to (signal) Ground. The switching on Pin2 corresponds to the surface mount LED, so will attach Pin2 to my micro as a basic "health" check. Useful.
